tests/component/test_stream_readers.py

```diff
@@ -545,3 +547,591 @@ def test___power_binary_reader___read_many_sample_with_wrong_dtype___raises_erro
+        )
+    return result
+
+
+def _bool_array_to_int(bool_array: numpy.ndarray) -> int:
+    result = 0
+    # Simulated data is little-endian
+    for bit in reversed(bool_array):
+        result = (result << 1) | bit
+    return result
+
+
+def test___digital_single_channel_reader___read_one_sample_one_line___returns_valid_samples(
+    di_single_line_task: nidaqmx.Task,
+) -> None:
+    reader = nidaqmx.stream_readers.DigitalSingleChannelReader(di_single_line_task.in_stream)
+    num_lines = di_single_line_task.channels.di_num_lines
+    samples_to_read = 10
+    data = [reader.read_one_sample_one_line() for _ in range(samples_to_read)]
+
+    assert data == _get_expected_digital_data(num_lines, samples_to_read)
+
+
+def test___digital_multi_channel_reader___read_one_sample_one_line___returns_valid_samples(
+    di_single_line_task: nidaqmx.Task,
+) -> None:
+    reader = nidaqmx.stream_readers.DigitalMultiChannelReader(di_single_line_task.in_stream)
+    num_lines = di_single_line_task.channels.di_num_lines
+    samples_to_read = 10
+    data = numpy.full(num_lines, False, dtype=numpy.bool_)
+
+    for sample_number in range(samples_to_read):
+        reader.read_one_sample_one_line(data)
+        assert _bool_array_to_int(data) == _get_expected_digital_data_for_sample(
+            num_lines, sample_number
+        )
+
+
+def test___digital_multi_channel_reader___read_one_sample_one_line_with_wrong_dtype___raises_error_with_correct_dtype(
+    di_single_line_task: nidaqmx.Task,
+) -> None:
+    reader = nidaqmx.stream_readers.DigitalMultiChannelReader(di_single_line_task.in_stream)
+    num_lines = di_single_line_task.channels.di_num_lines
+    data = numpy.full(num_lines, math.inf, dtype=numpy.float64)
+
+    with pytest.raises((ctypes.ArgumentError, TypeError)) as exc_info:
+        reader.read_one_sample_one_line(data)
+
+    assert "bool" in exc_info.value.args[0]
+
+
+def test___digital_single_channel_reader___read_one_sample_multi_line___returns_valid_samples(
+    di_single_channel_multi_line_task: nidaqmx.Task,
+) -> None:
+    reader = nidaqmx.stream_readers.DigitalSingleChannelReader(
+        di_single_channel_multi_line_task.in_stream
+    )
+    num_lines = di_single_channel_multi_line_task.channels.di_num_lines
+    samples_to_read = 10
+    data = numpy.full(num_lines, False, dtype=numpy.bool_)
+
+    for sample_number in range(samples_to_read):
+        reader.read_one_sample_multi_line(data)
+        assert _bool_array_to_int(data) == _get_expected_digital_data_for_sample(
+            num_lines, sample_number
+        )
+
+
+def test___digital_single_channel_reader___read_one_sample_multi_line_with_wrong_dtype___raises_error_with_correct_dtype(
+    di_single_channel_multi_line_task: nidaqmx.Task,
+) -> None:
+    reader = nidaqmx.stream_readers.DigitalSingleChannelReader(
+        di_single_channel_multi_line_task.in_stream
```

Comment:
[style] `from nidaqmx.stream_readers import DigitalSingleChannelReader, DigitalMultiChannelReader` would help some of these fit on one line.